src/Illuminate/Broadcasting/Broadcasters/AblyNativeBroadcaster.php

```diff
@@ -0,0 +1,343 @@
+     * @var int|mixed
+     * @default 1 hr
+     */
+    private $tokenExpiry = 3600;
+
+    /**
+     * Default channel capabilities, all public channels are by default given subscribe, history and channel-metadata access
+     * Set as per https://ably.com/docs/core-features/authentication#capability-operations.
+     *
+     * @var array
+     */
+    private $defaultChannelClaims = [
+        'public:*' => ['subscribe', 'history', 'channel-metadata'],
+    ];
+
+    /**
+     * Create a new broadcaster instance.
+     *
+     * @param  \Ably\AblyRest  $ably
+     * @param  array  $config
+     * @return void
+     */
+    public function __construct(AblyRest $ably, $config)
+    {
+        $this->ably = $ably;
+        if (self::$serverTimeDiff == null) {
+            self::setServerTime(round($this->ably->time() / 1000));
+        }
+        if (array_key_exists('disable_public_channels', $config) && $config['disable_public_channels']) {
+            $this->defaultChannelClaims = ['public:*' => ['channel-metadata']];
+        }
+        if (array_key_exists('token_expiry', $config)) {
+            $this->tokenExpiry = $config['token_expiry'];
+        }
+    }
+
+    private static $serverTimeDiff = null;
+
+    /**
+     * @param  int  $time
+     * @return void
+     */
+    private static function setServerTime($time)
+    {
+        self::$serverTimeDiff = time() - $time;
+    }
+
+    /**
+     * @return int
+     */
+    private static function getServerTime()
+    {
+        if (self::$serverTimeDiff != null) {
+            return time() - self::$serverTimeDiff;
+        }
+
+        return time();
+    }
+
+    /**
+     * Get the public token value from the Ably key.
+     *
+     * @return mixed
+     */
+    protected function getPublicToken()
+    {
+        return Str::before($this->ably->options->key, ':');
+    }
+
+    /**
+     * Get the private token value from the Ably key.
+     *
+     * @return mixed
+     */
+    protected function getPrivateToken()
+    {
+        return Str::after($this->ably->options->key, ':');
+    }
+
+    /**
+     * Authenticate the incoming request for a given channel.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return mixed
+     *
+     * @throws \Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException
+     */
+    public function auth($request)
+    {
+        $channelName = $request->channel_name;
+        $token = $request->token;
+        $connectionId = $request->socket_id;
+        $normalizedChannelName = $this->normalizeChannelName($channelName);
+        $userId = null;
+        $channelCapability = ['*'];
+        $user = $this->retrieveUser($request, $normalizedChannelName);
+        if ($user) {
+            $userId = method_exists($user, 'getAuthIdentifierForBroadcasting')
+                ? $user->getAuthIdentifierForBroadcasting()
+                : $user->getAuthIdentifier();
+        }
+        if ($this->isGuardedChannel($channelName)) {
```

Comment:
Always add some breathing room above and below an `if` statement.

```suggestion
        $user = $this->retrieveUser($request, $normalizedChannelName);

        if ($user) {
            $userId = method_exists($user, 'getAuthIdentifierForBroadcasting')
                ? $user->getAuthIdentifierForBroadcasting()
                : $user->getAuthIdentifier();
        }

        if ($this->isGuardedChannel($channelName)) {
```